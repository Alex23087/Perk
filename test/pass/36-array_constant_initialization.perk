import "stdio.h"

let ar: [int 3] = [2,3]

model ArrayConstantInitialization {
    let a: [int 3] = [42,89,3]
}

fun main(): void {
    let a: [int 3] = [4,5];
    let armodel := summon ArrayConstantInitialization();

    printf("a elements: %d, %d, %d\n", a[0], a[1], a[2]);
    printf("ar elements: %d, %d, %d\n", ar[0], ar[1], ar[2]);
    printf("amodel elements: %d, %d, %d\n", armodel.a[0], armodel.a[1], armodel.a[2]);
}