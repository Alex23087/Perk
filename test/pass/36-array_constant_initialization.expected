a elements: 4, 5, 0
ar elements: 2, 3, 0
amodel elements: 42, 89, 3
